core/templates/filters/translate.pipe.spec.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the translate pipe.
+ */
+
+import { async, TestBed, ComponentFixture, fakeAsync }
+  from '@angular/core/testing';
+import { EventEmitter, ChangeDetectorRef, Component } from '@angular/core';
+
+import { TranslateService } from 'services/translate.service';
+import { TranslatePipe } from './translate.pipe';
+import { UtilsService } from 'services/utils.service';
+
+
+class MockChangeDetectorRef {
+  markForCheck() {
+    return true;
+  }
+}
+
+class MockTranslateService {
+  langChangeEventEmitter = new EventEmitter();
+  get onLangChange() {
+    return this.langChangeEventEmitter;
+  }
+  translations = {
+    I18n_t_1: 'Hello',
+    I18n_t_2: 'Hello <[val]>',
+    I18n_rogue_1: '<script>alert(\'Oppia\');</script>Hello',
+    I18n_rogue_2: '<oppia-img>Me</oppia-img>Hola'
+  };
+
+  getInterpolatedString(key: string,
+      interpolateParams?: Object) {
+    const str = this.translations[key];
+    if (!str) {
+      return key;
+    }
+    return str;
+  }
+}
+
+describe('TranslatePipe', () => {
+  let pipe: TranslatePipe;
+  let translateService: TranslateService;
+  let changeDecRef: ChangeDetectorRef;
```

Comment:
Here and elsewhere, please avoid using abbreviations. Call it changeDetectorRef. Try to not have different-but-siimilar names for the same thing -- that's confusing.